inc/contract.class.php

```diff
@@ -1202,42 +1207,113 @@ static function cronContract($task = null) {
 
       foreach (Entity::getEntitiesToNotify('use_contracts_alert') as $entity => $value) {
          $before       = Entity::getUsedConfig('send_contracts_alert_before_delay', $entity);
-         $query_notice = "SELECT `glpi_contracts`.*
-                          FROM `glpi_contracts`
-                          LEFT JOIN `glpi_alerts`
-                              ON (`glpi_contracts`.`id` = `glpi_alerts`.`items_id`
-                                  AND `glpi_alerts`.`itemtype` = 'Contract'
-                                  AND `glpi_alerts`.`type`='".Alert::NOTICE."')
-                          WHERE (`glpi_contracts`.`alert` & ".pow(2, Alert::NOTICE).") >'0'
-                                AND `glpi_contracts`.`is_deleted` = 0
-                                AND `glpi_contracts`.`begin_date` IS NOT NULL
-                                AND `glpi_contracts`.`duration` <> 0
-                                AND `glpi_contracts`.`notice` <> 0
-                                AND DATEDIFF(ADDDATE(`glpi_contracts`.`begin_date`,
-                                                     INTERVAL `glpi_contracts`.`duration` MONTH),
-                                             CURDATE()) > 0
-                                AND DATEDIFF(ADDDATE(`glpi_contracts`.`begin_date`,
-                                                     INTERVAL (`glpi_contracts`.`duration`
-                                                                -`glpi_contracts`.`notice`) MONTH),
-                                             CURDATE()) < '$before'
-                                AND `glpi_alerts`.`date` IS NULL
-                                AND `glpi_contracts`.`entities_id` = '".$entity."'";
-
-         $query_end = "SELECT `glpi_contracts`.*
-                       FROM `glpi_contracts`
-                       LEFT JOIN `glpi_alerts`
-                           ON (`glpi_contracts`.`id` = `glpi_alerts`.`items_id`
-                               AND `glpi_alerts`.`itemtype` = 'Contract'
-                               AND `glpi_alerts`.`type`='".Alert::END."')
-                       WHERE (`glpi_contracts`.`alert` & ".pow(2, Alert::END).") > '0'
-                             AND `glpi_contracts`.`is_deleted` = 0
-                             AND `glpi_contracts`.`begin_date` IS NOT NULL
-                             AND `glpi_contracts`.`duration` <> '0'
-                             AND DATEDIFF(ADDDATE(`glpi_contracts`.`begin_date`,
-                                                  INTERVAL (`glpi_contracts`.`duration`) MONTH),
-                                          CURDATE()) < '$before'
-                             AND `glpi_alerts`.`date` IS NULL
-                             AND `glpi_contracts`.`entities_id` = '".$entity."'";
+
+         $query_notice = [
+            'SELECT'    => [
+               'glpi_contracts.*',
```

Comment:
useless?

Reply:
Not sure, request also includes alerts table and I do not know how ambiguous field names are handled by SQL and/or PHP.

Reply:
OK for me then